bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1026,21 +1026,24 @@ impl Client {
         &self,
         mime_type: String,
         data: Vec<u8>,
+        filename: Option<String>,
         progress_watcher: Option<Box<dyn ProgressWatcher>>,
     ) -> Result<String, ClientError> {
         let mime_type: mime::Mime = mime_type.parse().context("Parsing mime type")?;
-        let request = self.inner.media().upload(&mime_type, data, None);
+        let send_media_request = SendMediaUploadRequest::new((*self.inner).clone(), data)
+            .with_content_type(mime_type.essence_str().to_owned())
```

Comment:
It's surprising that `with_content_type()` doesn't take the same parameter type as the previous `&mime_type` in the previous call site. Can we make it so that it does, please?